src/fwrite.c

```diff
@@ -829,14 +820,18 @@ void fwriteMain(fwriteMainArgs args)
       DTPRINT(_("Allocate %zu bytes (%zu MiB) for zbuffPool\n"), alloc_size, alloc_size / MEGA);
     }
     zbuffPool = malloc(alloc_size);
-    if (!zbuffPool) {
+    if (zbuffPool == NULL) {
```

Comment:
We use this throughout the codebase and our "linter" checks for `!` usage specifically:

https://github.com/Rdatatable/data.table/blob/f72e46bbedc3f3f8c2abe0ea779d80e1e034ce32/.ci/linters/c/alloc_linter.R#L23

```suggestion
    if (!zbuffPool) {
```

Reply:
OK, sorry for this useless change.

Reply:
NP, it's of course equivalent, just a matter of style (and some guides recommend your approach as being more transparent). Only using `!` for consistency.